src/main/java/uk/gov/pay/adminusers/service/InviteService.java

```diff
@@ -141,16 +151,32 @@ public void generateOtp(InviteOtpRequest inviteOtpRequest) {
     }
 
     @Transactional
-    public User createInvitedUser(InviteValidateOtpRequest inviteValidateOtpRequest) {
+    public User createInvitedUserAndInvalidateInvite(InviteValidateOtpRequest inviteValidateOtpRequest) {
         Optional<InviteEntity> inviteOptional = inviteDao.findByCode(inviteValidateOtpRequest.getCode());
         if (inviteOptional.isPresent()) {
-            if (!secondFactorAuthenticator.authorize(inviteOptional.get().getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
-                throw invalidOtpAuthCodeInviteException(inviteValidateOtpRequest.getCode());
-            }
             InviteEntity inviteEntity = inviteOptional.get();
+            if (!secondFactorAuthenticator.authorize(inviteEntity.getOtpKey(), inviteValidateOtpRequest.getOtpCode())) {
+                // "failed login attempt" logic
+                inviteEntity.setLoginCounter(inviteEntity.getLoginCounter() + 1);
+                inviteEntity.setDisabled(inviteEntity.getLoginCounter() >= loginAttemptCap);
+                inviteDao.merge(inviteEntity);
+                // throw UNAUTHORIZED / 401 exception if the otp validation failed
+                throw invalidOtpAuthCodeInviteException(inviteEntity.getCode());
+            } else {
+                // "successful login attempt" logic
+                inviteEntity.setLoginCounter(0);
+            }
+            // check if the Invite is locked and throw UNAUTHORIZED / 401 exception if it is
+            if (inviteEntity.isDisabled()) {
+                throw inviteLockedException(inviteEntity.getCode());
```

Comment:
Once again, you should only throw exceptions from a `Transactional` methods if you know the rollback state is correct (basically you should know what you doing)